### PR TITLE
Develop branch adapts to openharmony

### PR DESCRIPTION
**Pull request by qiuguohua**
Re: #

### Changelog

Adaptation modules.
1、File
2、Audio
3、Network
4、Project
5、Font
etc.

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/core/settings.ts [hunk 1]
```diff
@@ -74,6 +75,11 @@ export class Settings {
             }
         }
         if (!path) return Promise.resolve();
+        if (window.oh) {
+            // TODO(qgh):OpenHarmony temporarily does not support reading json that is not in the resource directory
+            this._settings = require('../settings.json');
+            return Promise.resolve();
+        }
```
Comment (minggo) [inline cocos/core/settings.ts:82]: Does it mean that Settings is not available in OH?

File: native/CMakeLists.txt [hunk 2]
```diff
@@ -581,6 +597,20 @@ cocos_source_files(
...
+        cocos/platform/interfaces/modules/ISystemWindow.h
+        cocos/platform/openharmony/modules/System.cpp
+        cocos/platform/openharmony/modules/SystemWindow.cpp
+        cocos/platform/openharmony/modules/SystemWindowManager.h
+        cocos/platform/openharmony/modules/SystemWindowManager.cpp
+        cocos/platform/interfaces/modules/Device.cpp
+        cocos/platform/interfaces/modules/Device.h
+        cocos/platform/interfaces/modules/IScreen.h
+    )
+endif()
```
Comment (minggo) [inline native/CMakeLists.txt:613]: Why OH should use different header files?

File: cocos/core/settings.ts (hunk 1 above)
Comment (qiuguohua) [reply cocos/core/settings.ts:82]: It is supported.
Here is using another loading method

File: cocos/core/geometry/geometry-native-ext.ts [hunk 3]
```diff
@@ -87,7 +87,10 @@ const defineAttrFloat = (kls: Constructor, attr: string) => {
     // __nativeFields__ is defined in jsb_geometry_manual.cpp
     const desc: FieldDesc = (kls as any).__nativeFields__[attr];
     const cacheKey = `_$_${attr}`;
-    console.assert(desc.fieldSize === 4, `field ${attr} size ${desc.fieldSize}`);
+    if (!window.oh) {
+        // openharmony does not support the console.assert interface at this time.
+        console.assert(desc.fieldSize === 4, `field ${attr} size ${desc.fieldSize}`);
```
Comment (dumganhar) [inline cocos/core/geometry/geometry-native-ext.ts:92]: How about implementing the `console.assert` in ScriptEngine.cpp and export it to JS?
Then, this `if (!window.oh)` check could be removed.

File: native/CMakeLists.txt (hunk 2 above)
Comment (qiuguohua) [reply native/CMakeLists.txt:613]: It should be a very early adaptation legacy issue

File: native/cocos/bindings/manual/jsb_global.cpp [hunk 4]
```diff
@@ -92,11 +92,13 @@ static cc::network::Downloader *localDownloader() {
 }
 
 static void localDownloaderCreateTask(const ccstd::string &url, const std::function<void(const ccstd::string &, unsigned char *, int)> &callback) {
+#if CC_PLATFORM != CC_PLATFORM_OPENHARMONY // TODO(qgh):May be removed later
```
Comment (minggo) [inline native/cocos/bindings/manual/jsb_global.cpp:95]: It is better to describe the reason, then will not forget the reason in future.

File: native/cocos/bindings/manual/jsb_module_register.cpp [hunk 5]
```diff
@@ -79,10 +79,11 @@
     #include "cocos/bindings/manual/JavaScriptJavaBridge.h"
 #endif
 
-#if (CC_PLATFORM == CC_PLATFORM_IOS || CC_PLATFORM == CC_PLATFORM_ANDROID || CC_PLATFORM == CC_PLATFORM_OHOS)
-
-    #if CC_USE_VIDEO
-        #include "cocos/bindings/auto/jsb_video_auto.h"
+#if (CC_PLATFORM == CC_PLATFORM_IOS || CC_PLATFORM == CC_PLATFORM_ANDROID || CC_PLATFORM == CC_PLATFORM_OHOS || CC_PLATFORM == CC_PLATFORM_OPENHARMONY)
+    #if(CC_PLATFORM != CC_PLATFORM_OPENHARMONY)
```
Comment (minggo) [inline native/cocos/bindings/manual/jsb_module_register.cpp:83]: It is wired. Why not just don't include `CC_PLATFORM_OPENHARMONY ` in outer `#if`?

File: native/cocos/bindings/manual/jsb_module_register.cpp [hunk 6]
```diff
@@ -198,10 +199,11 @@ bool jsb_register_all_modules() {
...
 
-#if (CC_PLATFORM == CC_PLATFORM_IOS || CC_PLATFORM == CC_PLATFORM_ANDROID || CC_PLATFORM == CC_PLATFORM_OHOS)
-
-    #if CC_USE_VIDEO
-    se->addRegisterCallback(register_all_video);
+#if (CC_PLATFORM == CC_PLATFORM_IOS || CC_PLATFORM == CC_PLATFORM_ANDROID || CC_PLATFORM == CC_PLATFORM_OHOS || CC_PLATFORM == CC_PLATFORM_OPENHARMONY)
+    #if(CC_PLATFORM != CC_PLATFORM_OPENHARMONY)
+        #if CC_USE_VIDEO
+        se->addRegisterCallback(register_all_video);
+    #endif
```
Comment (minggo) [inline native/cocos/bindings/manual/jsb_module_register.cpp:206]: The same as above.

File: native/cocos/platform/openharmony/FileUtils-OpenHarmony.cpp [hunk 7]
```diff
@@ -0,0 +1,285 @@
+/****************************************************************************
+ Copyright (c) 2021-2022 Xiamen Yaji Software Co., Ltd.
```
Comment (minggo) [inline native/cocos/platform/openharmony/FileUtils-OpenHarmony.cpp:2]: 2021-2023?

File: platforms/native/engine/jsb-gfx.js [hunk 8]
```diff
@@ -57,7 +57,7 @@ deviceProto.copyTexImagesToTexture = function (texImages, texture, regions) {
 
 const oldDeviceCreateSwapchainFunc = deviceProto.createSwapchain;
 deviceProto.createSwapchain = function (info) {
-    info.windowHandle = jsbWindow.windowHandler;
+    info.windowHandle = window.oh ? jsb.device.getWindowHandle() : jsbWindow.windowHandler;
```
Comment (minggo) [inline platforms/native/engine/jsb-gfx.js:60]: Why oh is special?

File: scripts/build-engine/package-lock.json [hunk 9]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@cocos/build-engine",
-  "version": "4.4.1-alpha.10",
+  "version": "4.4.1-oh.0",
```
Comment (minggo) [inline scripts/build-engine/package-lock.json:3]: @PPpro could you please take a look this modification? Thanks.

Comment (PPpro) [reply scripts/build-engine/package-lock.json:3]: sure, it's OK here

Comment (minggo) [reply scripts/build-engine/package-lock.json:3]: ok

File: native/cocos/bindings/manual/jsb_global.cpp (hunk 4 above)
Comment (qiuguohua) [reply native/cocos/bindings/manual/jsb_global.cpp:95]: The network seems to be supported, I'll test it

File: native/cocos/bindings/manual/jsb_module_register.cpp (hunk 5 above)
Comment (qiuguohua) [reply native/cocos/bindings/manual/jsb_module_register.cpp:83]: This macro contains the webview and videoplay

File: platforms/native/engine/jsb-gfx.js (hunk 8 above)
Comment (qiuguohua) [reply platforms/native/engine/jsb-gfx.js:60]: Because openharmony does not support the execution of js scripts in c++

**Pull request closed without merging**